src/ppx_odoc_of_gospel/ppx_odoc_of_gospel.ml

```diff
@@ -0,0 +1,107 @@
+open Ppxlib
+
+let is_gospel attr = attr.attr_name.txt = "gospel"
+
+let is_ocaml_doc attr =
+  attr.attr_name.txt = "ocaml.text" || attr.attr_name.txt = "ocaml.doc"
+
+(* Prefix a string with whitespace the length of gospel special comment marker *)
+let align_gospel txt = "   " ^ txt
+
+let gospel_txt_of_attributes attr =
+  assert (is_gospel attr);
+  let rec aux attr =
+    if is_gospel attr then
+      match attr.attr_payload with
+      | PStr
+          [
+            {
+              pstr_desc =
+                Pstr_eval
+                  ( {
+                      pexp_desc = Pexp_constant (Pconst_string (txt, loc, _));
+                      _;
+                    },
+                    attrs );
+              _;
+            };
+          ] ->
+          { txt; loc } :: (List.map aux attrs |> List.flatten)
+      | _ -> []
+    else []
+  in
+  let rec squash (acc : string loc) = function
```

Comment:
Loops of string concatenations always make me uneasy, `String.concat` is a lot more efficient as it doesn’t unnecessarily churn the GC.